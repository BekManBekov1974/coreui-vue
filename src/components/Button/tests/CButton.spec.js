import { mount } from '@vue/test-utils'
import Component from '../CButton'

const ComponentName = 'CButton'
const defaultWrapper = mount(Component)
const toggleWrapper = mount(Component, {
  context: {
    props: {
      pressed: true,
      type: 'input',
      size: 'lg',
      variant: 'info',
      ghost: true,
      block: true,
      square: true
    }
  },
  slots: {
    default: 'button'
  }
})
const routerLinkWrapper = mount(Component, {
  context: {
    props: {
      to: '/dashboard',
      size: 'sm',
      variant: 'success',
      ghost: true,
      pill: true
    }
  },
  slots: {
    default: 'button'
  }
})

describe(ComponentName, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('renders correctly', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
  // it('renders correctly router link button', () => {
  //   expect(routerLinkWrapper.element).toMatchSnapshot()
  // })
  // it('renders correctly toggle button', () => {
  //   expect(toggleWrapper.element).toMatchSnapshot()
  // })
})
