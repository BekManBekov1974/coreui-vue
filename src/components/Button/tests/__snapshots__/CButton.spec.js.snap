// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CButton renders correctly 1`] = `
<button
  class="c-btn c-btn-primary"
  type="button"
/>
`;
